gradio/queueing.py

```diff
@@ -2,18 +2,28 @@
 
 import asyncio
 import copy
+import threading
 import time
 from asyncio import TimeoutError as AsyncTimeOutError
 from collections import deque
 from typing import Any
 
 import fastapi
 import httpx
-
-from gradio.data_classes import Estimation, PredictBody, Progress, ProgressUnit
+from typing_extensions import Literal
+
+from gradio.data_classes import (
+    Estimation,
+    LogMessage,
+    PredictBody,
+    Progress,
+    ProgressUnit,
+)
 from gradio.helpers import TrackedIterable
 from gradio.utils import AsyncRequest, run_coro_in_background, set_task_name
 
+thread_data = threading.local()
```

Comment:
I think we can store this in `Context` more consistent with how we share global variables and it feels more standard than storing a variable at the module level.

Reply:
done